src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -438,7 +438,22 @@ public virtual bool CanExecuteCode(string text)
 
         public Task<ExecutionResult> ResetAsync(bool initialize = true)
         {
-            GetInteractiveWindow().AddInput(_interactiveCommands.CommandPrefix + "reset");
+            // This is an implementation of ResetAsync on IInteractiveEvaluator interface.
+            // C# and VB Interactive window should not be calling this method at all
+            // as they now call reset through IInteractiveEvaluator2
+
+            // Now we pass on this call to IInteractiveEvaluation2 implementation
+            // which maintains old behaviour when isFromPrompt was not there
+            
+            return ResetAsync(initialize, true);
+        }
+
+        public Task<ExecutionResult> ResetAsync(bool initialize = true, bool isFromPrompt = true)
+        {
+            if (!isFromPrompt)
+            {
+                GetInteractiveWindow().AddInput(_interactiveCommands.CommandPrefix + "reset");
```

Comment:
@KevinH-MS this works most of the time but not always :-) 

It basically checks to see if the contentType in the current buffer is "interactive command" or not. So if you type #reset the content type changes to interactive command but if you hit the reset button at that point instead of hitting enter then inCommand will be true but the actual reset call is coming from the reset button.